### PR TITLE
Admin can del and create, edit user

### PR DESCRIPTION
**Pull request by tunv**
Admin can del user

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @user = User.new
+  end
+  
+  def index
+    @users = User.paginate page: params[:page]
+  end
+
+  def create
+    @user = User.new user_params
+    @user.supervisor = 1
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: Move 1 to Constant

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index]
+  before_action :admin_user,  only: [:new, :destroy, :edit]
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: Space

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  end
+
+  def edit
+    @user = User.find params[:id]
+  end
+
+  def update
+    @user = User.find params[:id]
+    if @user.update_attributes(user_params)
+      flash[:success] = t(:label, label: "Profile updated")
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: remove (), I18n

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      redirect_to @user
+    else
+      render "edit"
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: :edit

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    end
+  end
+
+  def destroy
+    @user = User.find params[:id]
+    @user.destroy
+    flash[:success] = t("Admin.user_delete")
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: ( )

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  private
+  def user_params
+    params.require(:user).permit :name, :email, :password, :password_confirmation
+  end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t(:label, label: :"Please log in.")
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: ( ), What is :"Please log in."

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -11,6 +11,7 @@ def new
   def create
     @user = User.new user_params
     if @user.save
+      log_in @user
       redirect_to @user
     else
       render "new"
```
Comment (leminhtuan2015) [inline app/controllers/users_controller.rb]: :new

File: app/views/admin/users/_user.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<ul class = "users">
+  <li>
+    <%= link_to user.name, [:admin, user] %>
+    <%= link_to t("Admin.edit"), edit_admin_user_path(user) %>
+    <%= link_to t("Admin.delete"), [:admin, user], method: :delete, 
+        data: {confirm: "Are you sure ?"} %>
```
Comment (leminhtuan2015) [inline app/views/admin/users/_user.html.erb]: I18n

File: app/views/admin/users/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("Admin.All_user") %>
+<h1><%= t("Admin.All_user") %></h1>
```
Comment (leminhtuan2015) [inline app/views/admin/users/index.html.erb]: ( )

File: app/views/admin/users/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<% provide :title, t("Admin.create") %>
```
Comment (leminhtuan2015) [inline app/views/admin/users/new.html.erb]: ( )

File: app/views/admin/users/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,5 @@
...
+<h1>Create User</h1>
```
Comment (leminhtuan2015) [inline app/views/admin/users/new.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 12]
```diff
@@ -1,6 +1,5 @@
 <% provide :title, t("signup") %>
```
Comment (leminhtuan2015) [inline app/views/users/new.html.erb]: ( )

File: app/views/admin/users/new.html.erb (hunk 10 above)
Comment (tunv) [inline app/views/admin/users/new.html.erb]: Không bỏ được cái này a ạ! Nó báo lỗi ạ!

File: app/views/admin/users/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+<% provide :title, t("Admin.All_user") %>
+<h1><%= t "Admin.All_user" %></h1>
+<%= will_paginate %>
+<ul class = "users">
+  <%= render @users %>
+</ul>
+<%= will_paginate %>
```
Comment (leminhtuan2015) [inline app/views/admin/users/index.html.erb]: Dubplicate

File: app/views/admin/users/new.html.erb (hunk 10 above)
Comment (leminhtuan2015) [inline app/views/admin/users/new.html.erb]: Sorry!!

**Comment by leminhtuan2015** [comment]
LGTM. Well done

File: app/controllers/admin/users_controller.rb [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+    if @user.save
+      log_in @user
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  def show
+    set_user
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: user `before_action`

File: app/controllers/admin/users_controller.rb [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+  end
+
+  def edit
+    set_user
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: use `before_action`

File: app/controllers/admin/users_controller.rb [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+  end
+  
+  def update
+    set_user
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: use `before_action`

File: app/controllers/admin/users_controller.rb [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    if @user.update_attributes(user_params)
+      flash[:success] = t "update_succes"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    set_user
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: use `before_action`

File: app/controllers/admin/users_controller.rb [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t "log_in_pl"
+      redirect_to login_url
+    end
+  end
+
+  def admin_user
+    redirect_to root_url unless current_user.supervisor?
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: flash message?

File: app/controllers/admin/users_controller.rb [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+
+  private
+  def user_params
+    params.require(:user).permit :name, :email, :password, :password_confirmation, :supervisor
+  end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t "log_in_pl"
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: don't use short word

File: app/controllers/admin/users_controller.rb [hunk 20]
```diff
@@ -0,0 +1,68 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index]
+  before_action :admin_user, only: [:new, :destroy, :edit]
+  before_action :set_user, except: [:new, :index]
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: about `create` action?

**Pull request merged**